src/odin_devices/si5324.py

```diff
@@ -0,0 +1,539 @@
+            + "{}\t\t{}\t\t{}\n".format("LOS2", self.Loss_Of_Signal_2_INT, self.Loss_Of_Signal_2_FLG) \
+            + "{}\t\t{}\t\t{}\n".format("LOSX", self.Loss_Of_Signal_X_INT, self.Loss_Of_Signal_X_FLG) \
+            + "{}\t\t{}\t\t{}\n".format("FO1", self.Freq_Offset_1_INT, self.Freq_Offset_1_FLG) \
+            + "{}\t\t{}\t\t{}\n".format("FO2", self.Freq_Offset_2_INT, self.Freq_Offset_2_FLG) \
+            )
+
+class SI5324(I2CDevice):
+    """
+    SI4324 Clock Multiplier Class:
+    """
+
+    # Registers that will require an iCAL calibration after modification
+    _ICAL_sensitive_registers = [0,1,2,4,5,7,7,9,10,11,19,25,31,34,40,43,46,55]
+    # Registers that should be included in the extracted register mapfile
+    _regmap_registers = [0,1,2,3,4,5,6,7,8,9,
+            10,11,19,
+            20,21,22,23,24,25,
+            31,32,33,34,35,36,
+            40,41,42,43,44,45,46,47,48,
+            55,
+            131,132,137,138,139,
+            142,143,
+            136]                        # Register 136 is here by convention (iCAL trigger)
+
+    # Clock IDs
+    CLOCK_NONE  = 0
+    CLOCK_1     = 1
+    CLOCK_2     = 2
+    CLOCK_X     = 3
+
+    # Autoselection Options:
+    AUTOMODE_Manual             = 0b00
+    AUTOMODE_Auto_Non_Revertive = 0b01
+    AUTOMODE_Auto_Revertive     = 0b10
+
+    # Define control fields within I2C registers
+    _FIELD_Free_Run_Mode = _Field(0,6,1)     #FREE_RUN Free Run Mode Enable
+
+    _FIELD_Clock_1_Priority = _Field(1,1,2)  #CK_PRIOR2 Clock with 2nd priority
+    _FIELD_Clock_2_Priority = _Field(1,3,2)  #CK_PRIOR1 Clock with 1st priority
+
+    _FIELD_Clock_Select = _Field(3,7,2)      #CLKSEL_REG Manual clock selection
+
+    _FIELD_Autoselection = _Field(4,7,2)     #AUTOSEL_REG Autoselection mode
+
+    _FIELD_Clock_Active = _Field(128,1,2)    #CKx_ACTV_REG for clocks 1 and 2
+
+    _FIELD_LOS1_INT = _Field(129,1,1)        #LOS1_INT Loss of Signal alarm for CLKIN_1
+    _FIELD_LOS2_INT = _Field(129,2,1)        #LOS2_INT Loss of Signal alarm for CLKIN_2
+    _FIELD_LOSX_INT = _Field(129,0,1)        #LOSX_INT Loss of Signal alarm for XA/XB
+
+    _FIELD_FOSC1_INT = _Field(130,1,1)       #FOSC1_INT Frequency Offset alarm for CLKIN_1
+    _FIELD_FOSC2_INT = _Field(130,2,1)       #FOSC2_INT Frequency Offset alarm for CLKIN_2
+    _FIELD_LOL_INT = _Field(130,0,1)         #LOL_INT Loss of Lock alarm
+
+    _FIELD_ICAL_TRG = _Field(136,6,1)        #ICAL Internal Calibration Trigger
+    _FIELD_RST_TRG = _Field(136,7,1)          #RST_REG Internal Reset Trigger
+
+    # NOTE: FLGs need manual clearing, for live alarm status, use corresponding INT signals...
+    _FIELD_FOSC1_FLG = _Field(132,2,1)       #FOSC1_FLG Frequency Offset Flag for CLKIN_1
+    _FIELD_FOSC2_FLG = _Field(132,3,1)       #FOSC2_FLG Frequency Offset Flag for CLKIN_2
+    _FIELD_LOL_FLG = _Field(132,1,1)         #LOL_FLG Loss of Lock Flag
+    # TODO define remaining relevant registers
```

Comment:
Outstanding TODO?

Reply:
There are quite a few fields from the datasheet I could include here. At the moment, the only ones present are the ones needed for the functions that have been included, but I wanted to make it clear where they could be added if more functionality was added later on. Instead of a TODO, would a comment along the lines of 'Additional fields should be specified here' be more appropriate?

Reply:
I think that's a good idea - doesn't give the impression that the code is incomplete.